plugins/soundsourcem4a/soundsourcem4a.cpp

```diff
@@ -430,27 +423,50 @@ SINT SoundSourceM4A::readSampleFrames(
     return samples2frames(numberOfSamplesTotal - numberOfSamplesRemaining);
 }
 
-} // namespace Mixxx
+QString SoundSourceProviderM4A::getName() const {
+    return "Nero FAAD2";
+}
 
-extern "C" MY_EXPORT const char* getMixxxVersion() {
-    return VERSION;
+QStringList SoundSourceProviderM4A::getSupportedFileExtensions() const {
+    QStringList supportedFileExtensions;
+    supportedFileExtensions.append("m4a");
+    supportedFileExtensions.append("mp4");
+    return supportedFileExtensions;
 }
 
-extern "C" MY_EXPORT int getSoundSourceAPIVersion() {
-    return MIXXX_SOUNDSOURCE_API_VERSION;
+namespace {
+
+void deleteSoundSource(SoundSource* pSoundSource) {
+    // The SoundSource must be deleted from within the external library
+    // that has allocated it.
+    delete pSoundSource;
 }
 
-extern "C" MY_EXPORT Mixxx::SoundSource* getSoundSource(QString fileName) {
-    return new Mixxx::SoundSourceM4A(fileName);
+} // anonymous namespace
+
+SoundSourcePointer SoundSourceProviderM4A::newSoundSource(const QUrl& url) {
+    return SoundSourcePointer(new SoundSourceM4A(url), deleteSoundSource);
 }
 
-extern "C" MY_EXPORT char** supportedFileExtensions() {
-    const QList<QString> supportedFileExtensions(
-            Mixxx::SoundSourceM4A::supportedFileExtensions());
-    return Mixxx::SoundSourcePlugin::allocFileExtensions(
-            supportedFileExtensions);
+} // namespace Mixxx
+
+namespace {
+
+// Singleton: SoundSourceProviderM4A is stateless
+// and a single instance can safely be shared
+Mixxx::SoundSourceProviderM4A SOUNDSOURCE_PROVIDER;
```

Comment:
In general, we should avoid static variables in libraries. 
Can't we make it a static local of Mixxx_SoundSourcePluginAPI_getSoundSourceProvider